src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs

```diff
@@ -0,0 +1,191 @@
+            var identifier = (IdentifierNameSyntax)identifierLocation.FindNode(cancellationToken);
+            var invocationOrCreation = (ExpressionSyntax)invocationOrCreationLocation.FindNode(cancellationToken);
+
+            var declaration = (VariableDeclarationSyntax)declarator.Parent;
+            if (declaration.Variables.Count == 1)
+            {
+                // Remove the entire declaration statement.
+                editor.RemoveNode(declaration.Parent);
+            }
+            else
+            {
+                // Otherwise, just remove the single declarator.
+                editor.RemoveNode(declarator);
+            }
+
+            var newType = this.GetDeclarationType(declaration.Type, useVarWhenDeclaringLocals, useImplicitTypeForIntrinsicTypes);
+
+            var declarationExpression = GetDeclarationExpression(identifier, newType);
+
+            var semanticsChanged = await SemanticsChangedAsync(
+                document, declaration, invocationOrCreation, newType,
+                identifier, declarationExpression, cancellationToken).ConfigureAwait(false);
+            if (semanticsChanged)
+            {
+                // Switching to 'var' changed semantics.  Just use the original type of the local.
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var local = (ILocalSymbol)semanticModel.GetDeclaredSymbol(declarator);
+
+                // If the user originally wrote it something other than 'var', then use what they
+                // wrote.  Otherwise, synthesize the actual type of the local.
+                var explicitType = declaration.Type.IsVar ? local.Type?.GenerateTypeSyntax() : declaration.Type;
+                declarationExpression = GetDeclarationExpression(identifier, explicitType);
+            }
+
+            editor.ReplaceNode(identifier, declarationExpression);
+        }
+
+        private static DeclarationExpressionSyntax GetDeclarationExpression(
+            IdentifierNameSyntax identifier, TypeSyntax newType)
+        {
+            newType = newType.WithoutTrivia().WithAdditionalAnnotations(Formatter.Annotation);
+            var declarationExpression = SyntaxFactory.DeclarationExpression(
+                SyntaxFactory.TypedVariableComponent(
+                    newType, SyntaxFactory.SingleVariableDesignation(identifier.Identifier)));
+            return declarationExpression;
+        }
+
+        private async Task<bool> SemanticsChangedAsync(
+            Document document,
+            VariableDeclarationSyntax declaration,
+            ExpressionSyntax invocationOrCreation,
+            TypeSyntax newType,
+            IdentifierNameSyntax identifier,
+            DeclarationExpressionSyntax declarationExpression,
+            CancellationToken cancellationToken)
+        {
+            if (newType.IsVar)
```

Comment:
Isn't there a helper that let's us determine if something is *really* var or not (i.e. not an alias or anything else)?